src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -4749,18 +4749,18 @@ namespace VariableSpeedCoils {
                         TotCapTempModFac =
                             Curve::CurveValue(state, varSpeedCoil.MSCCapFTemp(varSpeedCoil.NormSpedLevel), MixWetBulb, RatedSourceTempCool);
 
-                        if (MixTemp < SupTemp) {
-                            ShowSevereError(
-                                state,
-                                "On design day, cooling coil entering air temperature is less than design outlet air temperature. This will "
-                                "yield negative coil capacity sizing.");
-                            ShowContinueError(state, "Cooling capacity is set to zero during sizing; simulation continues.");
-                        }
                         if (MixEnth > SupEnth) {
                             CoolCapAtPeak = (rhoair * VolFlowRate * (MixEnth - SupEnth)) + FanCoolLoad;
                         } else {
                             CoolCapAtPeak = (rhoair * VolFlowRate * (48000.0 - SupEnth)) + FanCoolLoad;
                         }
+                        if ((CoolCapAtPeak < 0) && (MixTemp < SupTemp)) {
+                            ShowWarningError(
+                                state,
+                                format("On design day {}, cooling coil entering air temperature {:.2R} is less than design supply air temperature {:.2R}. This would "
+                                "yield negative coil capacity sizing for {}.", state.dataSize->FinalSysSizing(state.dataSize->CurSysNum).CoolDesDay, MixTemp, SupTemp, varSpeedCoil.Name));
+                            ShowContinueError(state, "Cooling capacity is set to zero during sizing; simulation continues.");
+                        }
```

Comment:
Coil sizing is based on enthalpy. Given the sneaky use of a standard capacity if enthalpies are not correct CoilCapAtPeak should never be negative here. I would like to see the 48000 removed (i.e., delete the else) since it just forces a non-zero coil capacity. There is nothing wrong with the coil sizing to 0 if FinalSysSizing conditions warrant that result.

Reply:
Yes! Not knowing where it came from, I was reticent to touch a "magic number."